packages/vue-app/template/components/nuxt-child.js

```diff
@@ -51,18 +51,25 @@ export default {
       if (beforeEnter) return beforeEnter.call(_parent, el)
     }
 
-    let routerView = [
-      h('router-view', data)
-    ]
+    // make sure that leave is called asynchronous (fix #5703)
+    if (transition.css === false) {
+      const leave = listeners.leave
+      listeners.leave = (el, done) => {
+        if (leave) leave.call(_parent, el)
+        return _parent.$nextTick(done)
+      }
+    }
+
+    let routerView = h('routerView', data)
+
     if (props.keepAlive) {
-      routerView = [
-        h('keep-alive', { props: props.keepAliveProps }, routerView)
-      ]
+      routerView = h('keep-alive', { props: props.keepAliveProps }, [routerView])
     }
+
     return h('transition', {
       props: transitionProps,
       on: listeners
-    }, routerView)
+    }, [routerView])
```

Comment:
Why do we need to pass children as an array? Is this a workaround or something?

Reply:
Not sure, but previously the routerView variable itself was an array. I changed that to this for better readability

Reply:
Fyi, this was without an array :)
```
Test Suites: 25 failed, 1 skipped, 74 passed, 99 of 100 total
Tests:       97 failed, 15 skipped, 655 passed, 767 total
```